src/Common/ThreadPool.cpp

```diff
@@ -767,6 +771,9 @@ void ThreadPoolImpl<Thread>::ThreadFromThreadPool::worker()
         if (DB::Exception::enable_job_stack_trace)
             DB::Exception::setThreadFramePointers(std::move(job_data->frame_pointers));
 
+        DB::Exception::parent_thread_id.store(job_data->parent_thread_id, std::memory_order_relaxed);
```

Comment:
I don't think we need `atomic_signal_fence` below and we can instead just use `memory_order_release` (both before and after)?
```suggestion
        DB::Exception::parent_thread_id.store(job_data->parent_thread_id, std::memory_order_release);
```

Reply:
we don't need neither fences nor memory ordering - because it's a single variable and accessed in the same thread but with concurrency from signal handler - so atomicity would suffice. So it needs a fix but in different place...

Reply:
Yeah that fix is even better.